packages/expect/src/matchers.ts

```diff
@@ -208,7 +227,11 @@ const matchers: MatchersObject = {
     return {message, pass};
   },
 
-  toBeGreaterThan(this: MatcherState, received: number, expected: number) {
+  toBeGreaterThan(
+    this: MatcherState,
+    received: number | bigint,
```

Comment:
@SimenB 
It will make Jest incompatible with TypeScript before 3.2. If bigint comes inside the exported `*.d.ts` old versions of TypeScript will refuse to compile.

I has the issue when I added the support for bigint in fast-check.

The possible workarounds are:
- tell users they need at least TypeScript 3.2
- or publish two different `*.d.ts` and use `typesVersions` in `package.json` (see https://github.com/dubzzz/fast-check/blob/master/package.json) - my choice in fast-check
- or ask users relying on old version of TypeScript to polyfill `bigint` for type system with for instance https://github.com/dubzzz/bigint-as-any-ts - by the way I finally chose option 2 instead of this one

Reply:
Great point, thanks! We can wait to land this until 25. Not sure if we've ever communicated a ts version though - this'll only affect people using the individual modules, not just `jest`

Reply:
@SimenB How about `types.ts` remind me does it provide intellisense for user test files?

Reply:
> only affect people using the individual modules, not just `jest`

You may want to check that before merging ;) I was quite surprised by the behaviour of TypeScript when I got this problem on fast-check. It failed even for users not using a function relying on `bigint`. `io-ts` lib had the same issue with `unknow` in the past.

Reply:
@SimenB - It would probably error out without importing any individual modules unless `skipLibCheck:true`   @dubzzz third bullet point would be simplest, not requiring any ts-versions but just making a note to use the polyfill.

Reply:
Easiest is to wait for the next major version, but it sucks to block features on this...

Reply:
The third choice is indeed the simplest one (and the one used by io-ts for `unknown`). I think it is not a bad choice and can allow old versions of TS to use latest Jest.

Second choice was pretty difficult and hacky to add.

The use of `skipLibCheck:true` is pretty dangerous. I saw this flag when I encoutered the issue but it removes lots of useful checks in TS :/

Reply:
But even on next major, your still going to have to communicated a ts version >=3.2 ?

Reply:
@SimenB Instead of blocking the feature, just put a note inside the release note explaining that users experiencing issues related to bigint - `Cannot find name 'bigint'` - with older versions of TS have to install an extra lib to polyfill ts.